### PR TITLE
Add flag to control number of mempool txns to sync in state syncer routine

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
* **#1083** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1083?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `feature/pos-networking-and-syncing`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lazynina and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cmd/config.go [hunk 1]
```diff
@@ -80,7 +80,8 @@ type Config struct {
 	TimeEvents            bool
 
 	// State Syncer
-	StateChangeDir string
+	StateChangeDir               string
+	NumMempoolTxStateSyncerLimit uint64
```
Comment (tholonious) [inline cmd/config.go:84]: Generally a good practice to prefix config names with the part of the system they apply to
```suggestion
	StateSyncerMempoolTxnSyncLimit uint64
```

File: cmd/config.go [hunk 2]
```diff
@@ -176,6 +177,7 @@ func LoadConfig() *Config {
 
 	// State Syncer
 	config.StateChangeDir = viper.GetString("state-change-dir")
+	config.NumMempoolTxStateSyncerLimit = viper.GetUint64("num-mempool-tx-state-syncer-limit")
```
Comment (tholonious) [inline cmd/config.go:180]:
```suggestion
	config.NumMempoolTxStateSyncerLimit = viper.GetUint64("state-syncer-mempool-txn-sync-limit")
```

File: lib/server.go [hunk 3]
```diff
@@ -409,6 +409,7 @@ func NewServer(
 	_augmentedBlockViewRefreshIntervalMillis uint64,
 	_posBlockProductionIntervalMilliseconds uint64,
 	_posTimeoutBaseDurationMilliseconds uint64,
+	_numMempoolTxStateSyncerLimit uint64,
```
Comment (tholonious) [inline lib/server.go:412]:
```suggestion
	_stateSyncerMempoolTxnSyncLimit uint64,
```

File: lib/state_change_syncer.go [hunk 4]
```diff
@@ -259,6 +259,8 @@ type StateChangeSyncer struct {
 	// of each entry, the consumer only has to sync the most recent version of each entry.
 	// BlocksyncCompleteEntriesFlushed is used to track whether this one time flush has been completed.
 	BlocksyncCompleteEntriesFlushed bool
+
+	NumMempoolTxSyncLimit uint64
```
Comment (tholonious) [inline lib/state_change_syncer.go:263]:
```suggestion
	MempoolTxnSyncLimit uint64
```

**Review by tholonious — APPROVED** [review]
Just some naming suggestions. LGTM otherwise

**Pull request merged**